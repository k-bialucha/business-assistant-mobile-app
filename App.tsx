import React from 'react';
import { StyledView, StyledText } from './App.styles';

const App = () => {
  return (
    <StyledView>
      <StyledText testID="welcome-text">
        Open up App.tsx to start working on your app!
      </StyledText>
    </StyledView>
  );
};

export default App;
