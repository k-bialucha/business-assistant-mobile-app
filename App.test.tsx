import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-native-testing-library';
import { shallow } from 'enzyme';

import App from './App';

describe('<App />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree.children.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = shallow(<App />);

    expect(tree).toMatchSnapshot();
  });

  it('contains welcome text', () => {
    const { getByTestId } = render(<App />);

    const element = getByTestId('welcome-text');

    expect(element.props.children.includes('start working on your app')).toBe(
      true
    );
  });
});
