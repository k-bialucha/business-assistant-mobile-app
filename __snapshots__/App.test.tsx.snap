// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<Styled(View)>
  <Styled(Text)
    testID="welcome-text"
  >
    Open up App.tsx to start working on your app!
  </Styled(Text)>
</Styled(View)>
`;
